packages/headless/src/modules/user-auth/services/user-profile.service.ts

```diff
@@ -0,0 +1,26 @@
+import { HttpStatus, Injectable } from '@nestjs/common';
+import { HelperService } from 'src/helper/helper.service';
+import { UserAuthRepository } from '../repository';
+
+@Injectable()
+export class UserProfileService {
+  constructor(
+    private userAuthRepo: UserAuthRepository,
+    private helperService: HelperService,
+  ) {}
+
+  async getProfile(userId: string) {
+    const doc = await this.userAuthRepo.findOne({ _id: userId });
```

Comment:
In the service Layer, You should not define {_id: userId}. Because we have several databases.
you can use ,`await this.userAuthRepo.findOne(userId);`

Reply:
refactored